sequencer/src/lib.rs

```diff
@@ -851,11 +868,56 @@ pub mod testing {
             self
         }
 
+        /// Version specific upgrade setup. Extend to future upgrades
+        /// by adding a branch to the `match` statement.
+        pub async fn set_upgrades(mut self, version: Version) -> Self {
+            let upgrade = match version {
+                version if version >= EpochVersion::VERSION => {
+                    tracing::debug!(?version, "upgrade version");
+                    let blocks_per_epoch = self.config.epoch_height;
+                    let epoch_start_block = self.config.epoch_start_block;
+
+                    let initial_stake_table =
+                        stake_table(self.config.known_nodes_with_stake.clone());
+
+                    let staking_private_keys =
+                        staking_priv_keys(&self.priv_keys, &self.state_key_pairs, NUM_NODES);
+
+                    let address = pos_deploy_routine(
```

Comment:
If we remove `stake_in_contract_for_test` from `pos_deploy_routine` and instead call it here then we don't need to pass `staking_priv_keys` to `pos_deploy_routine` and don't need to copy that code into the staking-cli crate.